src/transform/mod.rs

```diff
@@ -34,12 +39,14 @@ pub fn transform(rt: RichTerm) -> RichTerm {
         )
         .unwrap();
 
-    rt.traverse(
-        &mut |rt: RichTerm, _| -> Result<RichTerm, ()> {
-            let rt = share_normal_form::transform_one(rt);
+    rt.traverse_with_parent(
+        false,
+        &mut |rt: RichTerm, parent, (free_vars, field_free_vars)| -> Result<RichTerm, ()> {
+            let mut rt = share_normal_form::transform_one(rt);
+            collect_free_vars(&mut rt, parent, free_vars, field_free_vars);
```

Comment:
We probably want to do the transformation before `share_normal_form`, just because the latter introduces a bunch of let-binding and thus make the free variables calculation slower (while they don't change the free variables of recursive record fields).

Reply:
Switching their order produces an error.
Not sure why, any guesses?